### ISSUE TITLE
Prepare an API behavior document for all the APIs used by the app

### ISSUE THREAD

**Issue by maskaravivek** [commons-app/apps-android-commons#3385]
**Summary:** 

The commons app uses quite a few MediaWiki APIs throughout the app. It would be great to have an API collection with all the HTTP calls added there. 

It would facilitate in debugging issues quickly. We can quickly call and check an API to observe the response/latency or timeouts.

We can probably use https://www.postman.com/ or https://postwoman.io/ or some other framework for our API collection.

**Comment by GearGit** [comment]
@maskaravivek 
Can I work on this? It would improve my understanding of MediaWiki APIs

**Comment by maskaravivek** [comment]
Yes, if you wish to pick this up then you can start by creating a list of APIs with the parameters. We can later take a call to go with either postman or postwoman.

**Comment by GearGit** [comment]
> Yes, if you wish to pick this up then you can start by creating a list of APIs with the parameters. We can later take a call to go with either postman or postwoman.

The External API section in the documentation of the app contains a list of API used by the app. Is there any other API used whichbis not mentioned in the Docs? 
If yes then please do let me know, else I will create a list of API with their respective parameters. 

https://github.com/commons-app/apps-android-commons/wiki/External-APIs

**Comment by maskaravivek** [comment]
This doc was last updated in 2017 and a lot has changed since then. It would be great if you could play with the app and figure out what all APIs are used based on the logs. You can also look at retrofit interface classes to see the APIs.

**Comment by misaochan** [comment]
@maskaravivek I could not find such an option unfortunately. :/

**Comment by GearGit** [comment]
@maskaravivek
I am currently documenting API in Google docs. I am left with only few, how would you recommend me sharing the docs?

**Comment by maskaravivek** [comment]
You can just share the link here. Thanks

**Comment by GearGit** [comment]
@maskaravivek How would you recommend me to print the API in console as they are getting called? It is getting quite difficult to track API.

**Comment by maskaravivek** [comment]
You can go to OkHttpConnectionFactory and change the log level to `BODY` to see the full request-response logs.

**Comment by nicolas-raoul** [comment]
@GearGit I unassign for now, but if you are you still working on this, please let us know. If no answer, someone else may be assigned to it. Thanks a lot. :-)

**Comment by misaochan** [comment]
This would be a fantastic task for something like Outreachy, IMO. :)
